bin/pc2installsandbox

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+# Copyright (C) 1989-2023 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+#
+# Purpose: install a Linux cgroup v2 sandbox for PC2
+# Author: John Buck, based on original version by John Clevenger
+#
+
+FAIL_EXIT_CODE=43
+FAIL_BAD_ARGS_ALLOWED_EXIT_CODE=44
+FAIL_MISSING_CGROUP_CONTROLLERS_FILE=45
+FAIL_MISSING_CGROUP_SUBTREE_CONTROL_FILE=46
+FAIL_CPU_CONTROLLER_NOT_ENABLED=47
+FAIL_MEMORY_CONTROLLER_NOT_ENABLED=48
+FAIL_BAD_USER_GROUP=49
+
+CGROUP_PATH=/sys/fs/cgroup
+PC2_CGROUP_PATH=$CGROUP_PATH/pc2
+
+# ------------------------------------------------------------
+
+if [ "$#" -ne 2 ] ; then
+   echo $0: Unknown command line arguments: $*
+   echo Usage: $0 pc2user pc2group
+   exit $FAIL_BAD_ARGS_ALLOWED_EXIT_CODE
+fi 
+
+JUDGE_USER="$1"
+JUDGE_GROUP="$2"
+if ! getent passwd "$JUDGE_USER" >/dev/null 2>&1
+then
+   echo $0: No such user $JUDGE_USER.  The pc2user must be a valid login id.
+   exit $FAIL_BAD_USER_GROUP
+fi
+if ! getent group "$JUDGE_GROUP" >/dev/null 2>&1
+then
+   echo $0: No such group $JUDGE_USER.  The pc2group must be a valid group in /etc/group.
+   exit $FAIL_BAD_USER_GROUP
+fi
+
+# make sure CGroups v2 is installed
+echo Verifying CGroup v2 installation...
+if [ ! -f "$CGROUP_PATH/cgroup.controllers" ]; then
+   echo $0: "It appears CGroups v2 is not installed (missing file 'cgroup.controllers' in $CGROUP_PATH)."
+   echo CGroups v2 must be installed before a PC2 sandbox can be installed
+   echo "(or if CGroups v2 is installed elsewhere from $CGROUP_PATH, edit the CGroup Path in $0)."
+   exit $FAIL_MISSING_CGROUP_CONTROLLERS_FILE
+fi
+
+if [ ! -f "$CGROUP_PATH/cgroup.subtree_control" ]; then
+   echo $0: "It appears CGroups v2 is not installed (missing file 'cgroup.subtree_control' in $CGROUP_PATH)."
+   echo CGroups v2 must be installed before a PC2 sandbox can be installed
+   echo "(or if CGroups v2 is installed elsewhere from $CGROUP_PATH, edit the CGroup Path in $0)."
+   exit $FAIL_MISSING_CGROUP_SUBTREE_CONTROL_FILE
+fi
+
+# make sure the cpu and memory controllers are enabled
+
+if ! grep -q -F "cpu" "$CGROUP_PATH/cgroup.subtree_control"; then
+   echo $0: cgroup.subtree_control in $CGROUP_PATH does not enable cpu controller
+#   echo $0: Perhaps you have: echo +cpu ">$CGROUP_PATH/cgroup.subtree_control"
+   echo $0: Executing: echo +cpu ">$CGROUP_PATH/cgroup.subtree_control"
+   echo +cpu > $CGROUP_PATH/cgroup.subtree_control
+fi
+
+if ! grep -q -F "memory" "$CGROUP_PATH/cgroup.subtree_control"; then
+   echo $0: cgroup.subtree_control in $CGROUP_PATH does not enable memory controller
+   exit $FAIL_MEMORY_CONTROLLER_NOT_ENABLED
```

Comment:
Several lines above, the script tests for "cpu" and if it's not found then the script simply does an `echo +cpu` to the subtree_control.  I'm wondering why "memory" is handled by error-exiting instead of just adding it like "cpu" is handled?

Reply:
I have no idea.  I did that over 3 months ago.  I'll have to investigate why they are different.